computation_graph/graph.py

```diff
@@ -70,37 +91,41 @@ def make_computation_node(
         name=_infer_callable_name(func),
         func=func,
         is_stateful=_is_reducer_type(func),
-        signature=infer_callable_signature(func),
+        signature=_infer_callable_signature(func),
         is_terminal=False,
     )
 
 
 def make_edge(
-    source: Union[
-        Callable,
-        Tuple[Union[Callable, base_types.ComputationNode], ...],
-        base_types.ComputationNode,
-    ],
-    destination: Union[Callable, base_types.ComputationNode],
+    source: Union[_CallableOrNode, Tuple[_CallableOrNode, ...]],
+    destination: _CallableOrNode,
     key: Optional[Text] = None,
     priority: int = 0,
 ) -> base_types.ComputationEdge:
+    destination_as_node = make_computation_node(destination)
+    if destination_as_node.signature.is_kwargs:
+        assert not isinstance(source, tuple)
+        return base_types.ComputationEdge(
+            args=gamla.wrap_tuple(make_computation_node(source)),
```

Comment:
why is this going to args ? why not in source ? can you explain ?

Reply:
because we don't have a way to inject a nameless single variable. maybe you have a better idea?

Reply:
in the args case we use `key=None`
you are saying if we set `source=source, destination=destination, key=None` it doesn't work ?

Reply:
iirc the behaviour on `None` is to infer the name of the variable from the signature, which in this case cannot be done

Reply:
changed the implementation, ptal